ch02.adoc

```diff
@@ -268,3 +268,199 @@ They may not be attached to a group, even if all variables within that group use
+The order of elements in the **`substitutions`** attribute is not significant, and the substitutions for a given fragment must be such that applying them in any order will result in the same fragment dataset location.
+The use of substitutions can save space in the aggregation file; and in the event that the fragment locations need to be updated after the aggregation file has been created, it may be possible to achieve this by modifying the **`substitutions`** attribute rather than by changing the actual `location` fragment array variable values.
+See <<example-L.3>>.
+
+===== address
+
+The `address` fragment array variable, that may have any data type, defines how to find each fragment within its fragment dataset, i.e. the address of the fragment.
+In general it has the same dimensions in the same order as the `location` fragment array variable, and must contain a non-missing value corresponding to each fragment version.
+However, if the `address` fragment array variable is a scalar, then its single value applies to all versions of all fragments.
+For a netCDF fragment dataset, the address must be the string-valued netCDF variable name of the fragment.
+Addresses for other fragment dataset formats are allowed, on the understanding that an application program may choose to ignore any values that it does not understand.
+See <<example-L.1>> and <<example-L.6>>.
+
+===== value
+
+When the data values within a fragment are all the same, for each fragment, the `value` fragment array variable allows each fragment to be represented explicitly by its unique data value, rather than by reference to a fragment dataset.
+The `value` fragment array variable dimensions correspond to, and have the same sizes as, the fragment array dimensions in the same order as they appear in the conceptual fragment array.
+The `value` fragment array variable may have any data type, and contains each fragment's unique value.
+A fragment that contains wholly missing data is specified by any missing value indicated by the aggregation variable.
+See <<example-L.7>>, which uses an aggregation ancillary variable to make fragment dataset attributes available to an aggregation data variable.
+
+// Turn section numbering back on
+:numbered:
+
+
+[[fragment-interpretation, Section 2.8.2 "Fragment Interpretation"]]
+==== Fragment Interpretation
+
+The data of a fragment must be converted to its __canonical form__ prior to being inserted into the aggregated data. The canonical form of a fragment's data is such that:
+
+* The fragment's data, in its entirety, provide the values for a unique and contiguous part of the aggregated data.
+
+* The fragment's data dimensions correspond to the aggregated dimensions in the same order.
+
+* The fragment's data have the same units as the aggregation variable.
+
+* The fragment's data are not packed (i.e. not stored using a smaller data type than its original data).
```

Comment:
Here we are forbidding packing, but in the next paragraph, we seem to just be recommending against it (" It is therefore recommended that the aggregated data is not packed, because of the potential for mistakes and confusion."). We should choose which should be the case and make the text consistent in those two places. And perhaps mention the choice regarding packing in the conformance document.

Reply:
Unless we are talking about two different aspects, but I don't think so?

Reply:
We're not forbidding that a fragment is packed, rather we're saying that the fragment must be unpacked before being inserted into the aggregated data (just like we're saying that the fragment must have the same number of dimensions as the aggregated data, even if its data array in the file has fewer).

If the aggregation variable indicates packing then this, by definition, applies to the aggregated data - not to the individual fragments from which it is composed. In this case the _unpacked_ fragments are representing _packed_ values in the aggregated data. Hence the recommendation to _not_ have the aggregated data represent packed values - it's confusing and could lead to mistakes.

Reply:
Ah, I see now. OK, thanks for clarifying. I guess this is clear enough in the text now you state it, it is just a lot to get one's head around! Marking as resolved :)